### PR TITLE
Window Scaling and Settings Menu

### LINKED ISSUE #50: Resizable Window
Allow the player to resize the window to fit their screen. Hooooo boy this one might be a pain...

### PR DESCRIPTION
**Pull request by SuperShadowPlay**
Added:
- Basic window scaling
    - This is simply a multiplier on the base resolution of the game. While simple, it works well enough until we want to add some more complex HUD elements or allowing fullscreen resolution.
- A settings menu
    - Accessed by pressing the right arrow on the pause screen, it allows you to adjust screen resolution and game volume.
    
- Reworked a few internal menu systems
    - `shipMenu.lua` no longer has a duplicate copy of the blinking function
    - Menu key mapping has been simplified to where every menu has it's own `key_map` table, but it will fall back on the main menu's map if nothing is found.
 
Left for later:
- At this point, it may be useful to extend this new settings menu into a carousel that will allow for more "pages" of settings (mainly for keybinds).
- Allow player to wipe existing save data.

Closes #50
Partially addresses #64 (Blocked by #49)

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -20,7 +19,7 @@ local UpdateEntityAnimation = require('systems/UpdateEntityAnimation')
 local SpaceFriction = require('systems/SpaceFriction')
 
 love.load = function()
-    love.window.setMode(800, 600)
+    love.window.updateMode(State.camera.window_width, State.camera.window_height)
```
Comment (jaythomas) [inline main.lua:22]: Add `love.window.updateMode` to the mocks.

Comment (SuperShadowPlay) [reply main.lua:22]: Done

**Review by wyatt-herkamp — APPROVED** [review]
Hey, code looks good. I will test later.

**Review by Jon-Cavaliere — APPROVED** [review]
Works well, code makes sense

**Pull request merged**